lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1418 @@
+  }
+
+  get bcnlSerialNumber() {
+    return this.bcnlModelAndSerial.replace(/\d+-/, '');
+  }
+
+  get iv() {
+    const iv = Buffer.from(this.key);
+    iv[0] = this.radioChannel;
+
+    return iv;
+  }
+
+  get packedLinkMAC() {
+    return Buffer.from(this.linkMAC, 'hex').swap64().toString('binary');
+  }
+
+  get packedPumpMAC() {
+    return Buffer.from(this.pumpMAC, 'hex').swap64().toString('binary');
+  }
+
+  getHMAC() {
+    const paddingKey = 'A4BD6CED9A42602564F413123';
+    const digest = crypto.createHash('sha256')
+      .update(`${this.bcnlSerialNumber}${paddingKey}`)
+      .digest()
+      .reverse();
+    return digest;
+  }
+}
+
+class MinimedMessage extends BCNLMessage {
+  static get COMMAND_TYPE() {
+    return {
+      OPEN_CONNECTION: 0x10,
+      CLOSE_CONNECTION: 0x11,
+      SEND_MESSAGE: 0x12,
+      READ_INFO: 0x14,
+      REQUEST_LINK_KEY: 0x16,
+      SEND_LINK_KEY: 0x17,
+      RECEIVE_MESSAGE: 0x80,
+      SEND_MESSAGE_RESPONSE: 0x81,
+      REQUEST_LINK_KEY_RESPONSE: 0x86,
+    };
+  }
+
+  static get ENVELOPE_SIZE() {
+    return 33;
+  }
+
+  static get MINIMED_HEADER() {
+    return 0x5103; // Q\x03
+  }
+
+  static oneByteChecksum(buffer) {
+    // eslint-disable-next-line no-bitwise
+    return _.reduce(buffer, (a, b) => a + b, 0) & 0xff;
+  }
+}
+
+class MinimedResponse extends MinimedMessage {
+  constructor(payload) {
+    super(payload);
+
+    if (this.payload.readUInt16BE(0x00) !== MinimedMessage.MINIMED_HEADER) {
+      throw new InvalidMessageError('Unexpected MiniMed packet header.');
+    }
+
+    const minimedPayloadSize = this.payload.readUInt16LE(0x1C);
+    // Check the payload's checksums
+    const minimedPayload =
+      Buffer.from(this.payload.slice(0x00, MinimedMessage.ENVELOPE_SIZE + minimedPayloadSize));
+    const expectedChecksum = minimedPayload[0x20];
+    const calculatedChecksum =
+      // eslint-disable-next-line no-bitwise
+      (MinimedMessage.oneByteChecksum(minimedPayload) - expectedChecksum) & 0xFF;
+
+    if (expectedChecksum !== calculatedChecksum) {
+      throw new ChecksumError(expectedChecksum, calculatedChecksum);
+    }
+  }
+}
+
+class MinimedRequest extends MinimedMessage {
+  constructor(commandType, pumpSession, payload, responseType = MinimedResponse) {
+    if (new.target === MinimedRequest) {
+      throw new TypeError(`Cannot construct ${new.target.name} instances directly`);
+    }
+
+    super(MinimedRequest.buildPayload(commandType, pumpSession, payload), responseType);
+    this.pumpSession = pumpSession;
+  }
+
+  static buildPayload(commandType, pumpSession, payload) {
+    const payloadLength = (payload === null) ? 0 : payload.length;
+    const payloadBuffer = Buffer.alloc(MinimedMessage.ENVELOPE_SIZE + payloadLength);
+
+    /* eslint-disable lodash/prefer-lodash-method */
+    payloadBuffer.writeUInt16BE(MinimedMessage.MINIMED_HEADER, 0); // Q\x03
+    payloadBuffer.write('000000', 2); // Pump serial. '000000' for 600-series
```

Comment:
Yes, that's correct. The driver gets the serial number of the pump from the paired BCN Link. 
The pump can be paired with multiple BCNL's, but a BCNL can only be paired with one pump.
It's also possible to create an "ad hoc" pairing, but after discussion with Brandon, we're sticking with the "already paired" model.